// @progress/kendo-angular-upload/dist/npm/upload-events.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @hidden
 */
var PreventableEvent = (function () {
    function PreventableEvent() {
        this.prevented = false;
    }
    /**
     * Prevents the default action for a specified event.
     * In this way, the source component suppresses the built-in behavior that follows the event.
     */
    PreventableEvent.prototype.preventDefault = function () {
        this.prevented = true;
    };
    /**
     * If the event is prevented by any of its subscribers, returns `true`.
     *
     * @returns `true` if the default action was prevented. Otherwise, returns `false`.
     */
    PreventableEvent.prototype.isDefaultPrevented = function () {
        return this.prevented;
    };
    return PreventableEvent;
}());
exports.PreventableEvent = PreventableEvent;
/**
 * {% embed_file events/cancel/upload.component.ts preview %}
 * {% embed_file events/cancel/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 *
 * Fires when the user cancels the process of uploading a file or a batch of files.
 *
 * Arguments for the `cancel` event:
 */
var CancelEvent = (function () {
    /**
     * Constructs the event arguments for the `cancel` event.
     * @param files - The list of the files that were to be uploaded.
     */
    function CancelEvent(files) {
        this.files = files;
    }
    return CancelEvent;
}());
exports.CancelEvent = CancelEvent;
/**
 * {% embed_file events/clear/upload.component.ts preview %}
 * {% embed_file shared/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 *
 * Fires when the **Clear** button is clicked. At this point, the selected files are about to be cleared.
 */
var ClearEvent = (function (_super) {
    __extends(ClearEvent, _super);
    /**
     * Constructs the event arguments for the `clear` event.
     */
    function ClearEvent() {
        return _super.call(this) || this;
    }
    return ClearEvent;
}(PreventableEvent));
exports.ClearEvent = ClearEvent;
/**
 * {% embed_file events/error/upload.component.ts preview %}
 * {% embed_file shared/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 *
 * Fires when an upload or removal operation fails.
 *
 * Arguments for the `error` event:
 */
var ErrorEvent = (function () {
    /**
     * Constructs the event arguments for the `error` event.
     *
     * @param files - The list of the files that failed to be uploaded or removed.
     * @param operation - The operation type (`upload` or `remove`).
     * @param response - The response object returned by the server.
     */
    function ErrorEvent(files, operation, response) {
        this.files = files;
        this.operation = operation;
        this.response = response;
    }
    return ErrorEvent;
}());
exports.ErrorEvent = ErrorEvent;
/**
 * {% embed_file events/remove/upload.component.ts preview %}
 * {% embed_file shared/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 *
 * Fires when an uploaded or selected file is about to be removed. If you cancel the event, the removal is prevented.
 *
 * Arguments for the `remove` event:
 */
var RemoveEvent = (function (_super) {
    __extends(RemoveEvent, _super);
    /**
     * Constructs the event arguments for the `remove` event.
     * @param files - The list of the files that are about to be removed.
     * @param headers - The headers of the request.
     */
    function RemoveEvent(files, headers) {
        var _this = _super.call(this) || this;
        _this.files = files;
        _this.headers = headers;
        return _this;
    }
    return RemoveEvent;
}(PreventableEvent));
exports.RemoveEvent = RemoveEvent;
/**
 * {% embed_file events/select/upload.component.ts preview %}
 * {% embed_file shared/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 *
 * Fires when the user selects a file or multiple files for upload. If you cancel the event, the selection is prevented.
 *
 * Arguments for the `select` event:
 */
var SelectEvent = (function (_super) {
    __extends(SelectEvent, _super);
    /**
     * Constructs the event arguments for the `select` event.
     * @param files - The list of the selected files.
     */
    function SelectEvent(files) {
        var _this = _super.call(this) || this;
        _this.files = files;
        return _this;
    }
    return SelectEvent;
}(PreventableEvent));
exports.SelectEvent = SelectEvent;
/**
 * {% embed_file events/success/upload.component.ts preview %}
 * {% embed_file shared/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 *
 * Fires when the selected files are successfully uploaded or removed.
 *
 * Arguments for the `success` event:
 */
var SuccessEvent = (function (_super) {
    __extends(SuccessEvent, _super);
    /**
     * Constructs the event arguments for the `success` event.
     * @param files - The list of the files that were uploaded or removed.
     * @param operation - The operation type (`upload` or `remove`).
     * @param response - The response object returned by the server.
     */
    function SuccessEvent(files, operation, response) {
        var _this = _super.call(this) || this;
        _this.files = files;
        _this.operation = operation;
        _this.response = response;
        return _this;
    }
    return SuccessEvent;
}(PreventableEvent));
exports.SuccessEvent = SuccessEvent;
/**
 * Fires when one or more files are about to be uploaded. If you cancel the event, the upload is prevented.
 * Headers can be added to the request:
 * {% embed_file events/upload/upload.component.ts preview %}
 * {% embed_file shared/app.component.ts %}
 * {% embed_file shared/main.ts hidden %}
 * Arguments for the `upload` event:
 */
var UploadEvent = (function (_super) {
    __extends(UploadEvent, _super);
    /**
     * Constructs the event arguments for the `upload` event.
     * @param files - The list of the files that are about to be uploaded.
     * @param headers - The headers of the request.
     */
    function UploadEvent(files, headers) {
        var _this = _super.call(this) || this;
        _this.files = files;
        _this.headers = headers;
        return _this;
    }
    return UploadEvent;
}(PreventableEvent));
exports.UploadEvent = UploadEvent;
