// @progress/kendo-angular-upload/dist/npm/upload.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var file_info_1 = require("./file-info");
var file_map_1 = require("./file-map");
var http_1 = require("@angular/http");
var upload_events_1 = require("./upload-events");
var util_1 = require("./util");
/**
 * @hidden
 */
var UploadService = (function () {
    function UploadService(http) {
        this.http = http;
        this.cancelEvent = new core_1.EventEmitter();
        this.clearEvent = new core_1.EventEmitter();
        this.completeEvent = new core_1.EventEmitter();
        this.errorEvent = new core_1.EventEmitter();
        this.removeEvent = new core_1.EventEmitter();
        this.selectEvent = new core_1.EventEmitter();
        this.successEvent = new core_1.EventEmitter();
        this.uploadEvent = new core_1.EventEmitter();
        //required for ControlValueAccessor integration
        this.changeEvent = new core_1.EventEmitter();
        this.fileList = new file_map_1.FileMap();
    }
    Object.defineProperty(UploadService.prototype, "files", {
        get: function () {
            return this.fileList;
        },
        enumerable: true,
        configurable: true
    });
    UploadService.prototype.onChange = function () {
        var files = this.fileList.filesFlat.filter(function (file) {
            return file.state === file_info_1.FileState.Initial ||
                file.state === file_info_1.FileState.Uploaded;
        });
        this.changeEvent.emit(files.length > 0 ? files : null);
    };
    UploadService.prototype.addFiles = function (files, async) {
        var selectEventArgs = new upload_events_1.SelectEvent(files);
        this.selectEvent.emit(selectEventArgs);
        if (!selectEventArgs.isDefaultPrevented()) {
            for (var _i = 0, files_1 = files; _i < files_1.length; _i++) {
                var file = files_1[_i];
                this.fileList.add(file);
            }
            if (async.autoUpload) {
                this.uploadFiles(async);
            }
        }
    };
    UploadService.prototype.addInitialFiles = function (initialFiles) {
        var _this = this;
        this.fileList.clear();
        initialFiles.forEach(function (file) {
            var fakeFile = util_1.default.getInitialFileInfo(file);
            _this.fileList.add(fakeFile);
        });
    };
    UploadService.prototype.removeFiles = function (uid, async) {
        var removedFiles = this.fileList.get(uid);
        // cloning the Headers so the default ones are not overridden
        var removeEventArgs = new upload_events_1.RemoveEvent(removedFiles, new http_1.Headers(async.removeHeaders));
        this.removeEvent.emit(removeEventArgs);
        if (!removeEventArgs.isDefaultPrevented()) {
            if (removedFiles[0].state === file_info_1.FileState.Uploaded ||
                removedFiles[0].state === file_info_1.FileState.Initial) {
                this.performRemove(removedFiles, async, removeEventArgs);
            }
            else {
                this.fileList.remove(uid);
            }
        }
    };
    UploadService.prototype.cancelFiles = function (uid) {
        var canceledFiles = this.fileList.get(uid);
        var cancelEventArgs = new upload_events_1.CancelEvent(canceledFiles);
        this.cancelEvent.emit(cancelEventArgs);
        for (var _i = 0, canceledFiles_1 = canceledFiles; _i < canceledFiles_1.length; _i++) {
            var file = canceledFiles_1[_i];
            if (file.httpSubscription) {
                file.httpSubscription.unsubscribe();
            }
        }
        this.fileList.remove(uid);
    };
    UploadService.prototype.clearFiles = function () {
        var clearEventArgs = new upload_events_1.ClearEvent();
        this.clearEvent.emit(clearEventArgs);
        if (!clearEventArgs.isDefaultPrevented()) {
            var triggerChange = this.fileList.hasFileWithState([file_info_1.FileState.Initial, file_info_1.FileState.Uploaded]);
            this.fileList.clear();
            if (triggerChange) {
                this.onChange();
            }
        }
    };
    UploadService.prototype.uploadFiles = function (async) {
        var allFilesToUpload = this.fileList.filesToUpload;
        this._uploadFiles(allFilesToUpload, async);
    };
    UploadService.prototype.retryFiles = function (uid, async) {
        var filesToRetry = [this.fileList.get(uid)];
        this._uploadFiles(filesToRetry, async);
    };
    UploadService.prototype._uploadFiles = function (allFiles, async) {
        var _loop_1 = function (filesToUpload) {
            // cloning the Headers so the default ones are not overridden
            var uploadEventArgs = new upload_events_1.UploadEvent(filesToUpload, new http_1.Headers(async.saveHeaders));
            this_1.uploadEvent.emit(uploadEventArgs);
            if (!uploadEventArgs.isDefaultPrevented()) {
                this_1.fileList.setFilesState(filesToUpload, file_info_1.FileState.Uploading);
                var httpSubcription_1 = this_1.performUpload(filesToUpload, async, uploadEventArgs);
                filesToUpload.forEach(function (file) {
                    file.httpSubscription = httpSubcription_1;
                });
            }
            else {
                this_1.fileList.remove(filesToUpload[0].uid);
            }
        };
        var this_1 = this;
        for (var _i = 0, allFiles_1 = allFiles; _i < allFiles_1.length; _i++) {
            var filesToUpload = allFiles_1[_i];
            _loop_1(filesToUpload);
        }
    };
    UploadService.prototype.performRemove = function (files, async, removeEventArgs) {
        var _this = this;
        var fileNames = files.map(function (file) {
            return file.name;
        });
        var formData = this.populateRemoveFormData(fileNames, async.removeField, removeEventArgs.data);
        var options = this.populateRequestOptions(async, formData, removeEventArgs.headers, "remove");
        var removeRequest = new http_1.Request(options);
        this.http.request(removeRequest)
            .subscribe(function (success) {
            _this.onSuccess(success, files, "remove");
        }, function (error) {
            _this.onError(error, files, "remove");
        });
    };
    UploadService.prototype.performUpload = function (files, async, uploadEventArgs) {
        var _this = this;
        var formData = this.populateUploadFormData(files, async.saveField, uploadEventArgs.data);
        var options = this.populateRequestOptions(async, formData, uploadEventArgs.headers, "upload");
        var uploadRequest = new http_1.Request(options);
        var httpSubscription = this.http.request(uploadRequest)
            .subscribe(function (success) {
            _this.onSuccess(success, files, "upload");
            _this.checkAllComplete();
        }, function (error) {
            _this.onError(error, files, "upload");
            _this.checkAllComplete();
        });
        return httpSubscription;
    };
    UploadService.prototype.onSuccess = function (successResponse, files, operation) {
        var successArgs = new upload_events_1.SuccessEvent(files, operation, successResponse);
        this.successEvent.emit(successArgs);
        if (operation === "upload") {
            this.fileList.setFilesState(files, successArgs.isDefaultPrevented() ? file_info_1.FileState.Failed : file_info_1.FileState.Uploaded);
        }
        else {
            if (!successArgs.isDefaultPrevented()) {
                this.fileList.remove(files[0].uid);
            }
        }
        if (!successArgs.isDefaultPrevented()) {
            this.onChange();
        }
    };
    UploadService.prototype.onError = function (errorResponse, files, operation) {
        var errorArgs = new upload_events_1.ErrorEvent(files, operation, errorResponse);
        this.errorEvent.emit(errorArgs);
        if (operation === "upload") {
            this.fileList.setFilesState(files, file_info_1.FileState.Failed);
        }
    };
    UploadService.prototype.checkAllComplete = function () {
        if (!this.fileList.hasFileWithState([file_info_1.FileState.Selected, file_info_1.FileState.Uploading])) {
            this.completeEvent.emit();
        }
    };
    UploadService.prototype.populateRequestOptions = function (async, formData, headers, operation) {
        var options = new http_1.RequestOptions({
            body: formData,
            method: (operation === "upload") ? async.saveMethod : async.removeMethod,
            url: (operation === "upload") ? async.saveUrl : async.removeUrl,
            withCredentials: async.withCredentials
        });
        if (headers.keys().length > 0) {
            options.headers = headers;
        }
        return options;
    };
    UploadService.prototype.populateUploadFormData = function (files, saveField, clientData) {
        var data = new FormData();
        this.populateClientFormData(data, clientData);
        for (var _i = 0, files_2 = files; _i < files_2.length; _i++) {
            var file = files_2[_i];
            data.append(saveField, file.rawFile);
        }
        return data;
    };
    UploadService.prototype.populateRemoveFormData = function (fileNames, removeField, clientData) {
        var data = new FormData();
        this.populateClientFormData(data, clientData);
        for (var _i = 0, fileNames_1 = fileNames; _i < fileNames_1.length; _i++) {
            var fileName = fileNames_1[_i];
            data.append(removeField, fileName);
        }
        return data;
    };
    UploadService.prototype.populateClientFormData = function (data, clientData) {
        for (var key in clientData) {
            if (clientData.hasOwnProperty(key)) {
                data.append(key, clientData[key]);
            }
        }
    };
    return UploadService;
}());
UploadService.decorators = [
    { type: core_1.Injectable },
];
/** @nocollapse */
UploadService.ctorParameters = function () { return [
    { type: http_1.Http, },
]; };
exports.UploadService = UploadService;
