// @progress/kendo-angular-upload/dist/es/file-list-item-base.js
import util from "./util";
/**
 * @hidden
 */
var FileListItemBase = (function () {
    function FileListItemBase() {
    }
    FileListItemBase.prototype.fileHasValidationErrors = function (file) {
        return util.fileHasValidationErrors(file);
    };
    FileListItemBase.prototype.filesHaveValidationErrors = function (files) {
        return util.filesHaveValidationErrors(files);
    };
    FileListItemBase.prototype.getFileValidationMessage = function (file) {
        var validationMessage;
        if (file.validationErrors && file.validationErrors.length > 0) {
            validationMessage = this.localization.get(file.validationErrors[0]);
        }
        return validationMessage;
    };
    FileListItemBase.prototype.getInvalidFilesMessage = function () {
        return this.localization.get('invalidFiles');
    };
    FileListItemBase.prototype.getTotalFilesSizeMessage = function (files) {
        return util.getTotalFilesSizeMessage(files);
    };
    return FileListItemBase;
}());
export { FileListItemBase };
